packages/core/src/modules/credentials/services/CredentialService.ts

```diff
@@ -247,16 +247,29 @@ export abstract class CredentialService {
     return this.credentialRepository.findById(connectionId)
   }
 
-  public async deleteById(credentialId: string, options?: DeleteCredentialOptions): Promise<void> {
-    const credentialRecord = await this.getById(credentialId)
+  /**
+   * Delete a credential exchange record from the repository and (optionally) 
+   * the wallet.
+   * Note that the default is now to delete from the wallet
+   * 
+   * @param credentialRecordId The id of the credential record
+   * @param options contains flag for determining whether to delete from wallet as well as repository
+
+   */
+  public async deleteById(credentialRecordId: string, options?: DeleteCredentialOptions): Promise<void> {
+    const credentialRecord = await this.getById(credentialRecordId)
 
     await this.credentialRepository.delete(credentialRecord)
 
-    if (options?.deleteAssociatedCredentials) {
-      for (const credential of credentialRecord.credentials) {
-        const formatService: CredentialFormatService = this.getFormatService(credential.credentialRecordType)
-        await formatService.deleteCredentialById(credentialRecord, options)
-      }
+    let deleteOptions = options
+
+    if (!deleteOptions || !deleteOptions.deleteAssociatedCredentials) {
+      deleteOptions = { ...options, deleteAssociatedCredentials: true }
+    }
+
+    for (const credential of credentialRecord.credentials) {
+      const formatService: CredentialFormatService = this.getFormatService(credential.credentialRecordType)
+      await formatService.deleteCredentialById(credentialRecord, deleteOptions)
```

Comment:
Sorry -- I should have taken better notice the first review. But why do we need to provide the deleteAssociatedCredential parameter to the formatService? The format service should only be called if the associated credential needs to be deleted.

```
const deleteAssociatedCredentials = options?.deleteAssociatedCredentials ?? true

if (deleteAssociatedCredentials) {
    for (const credential of credentialRecord.credentials) {
      const formatService = this.getFormatService(credential.credentialRecordType)
     // this takes the `credential.credentialId`, not the credentialRecord!
      await formatService.deleteCredentialById(credential.credentialId)
  }
}
```